src/functional/ffront/foast_to_itir.py

```diff
@@ -157,11 +211,34 @@ def visit_Call(self, node: foast.Call, **kwargs) -> itir.FunCall:
         if TypeInfo(node.func.type).is_field_type:
             return self._visit_shift(node, **kwargs)
         elif node.func.id in FUN_BUILTIN_NAMES:
-            return self._visit_reduce(node, **kwargs)
-        return self._lift_lambda(node)(
+            visitor = getattr(self, f"_visit_{node.func.id}")
+            return visitor(node, **kwargs)
+        elif node.func.id in TYPE_BUILTIN_NAMES:
+            return self._visit_type_constr(node, **kwargs)
+        return self._lift_if_field(node)(
             im.call_(self.visit(node.func, **kwargs))(*self.visit(node.args, **kwargs))
         )
 
+    def _visit_neighbor_sum(self, node: foast.Call, **kwargs) -> itir.FunCall:
+        return self._visit_reduce(node, **kwargs)
+
+    def _visit_type_constr(self, node: foast.Call, **kwargs) -> itir.Literal:
+        if isinstance(node.args[0], foast.Constant):
+            target_type = fbuiltins.BUILTINS[node.func.id]
+            source_type = {**fbuiltins.BUILTINS, "string": str}[
```

Comment:
```suggestion
            target_type = fbuiltins.TYPE_BUILTINS[node.func.id]
            source_type = {**fbuiltins.TYPE_BUILTINS, "string": str}[
```

Reply:
The suggestion would break because `fbuiltins.TYPE_BUILTINS` is not a dictionary.

Reply:
I see, since the caller restricts node.func.id I'm fine with this.